File: 28_find_the_index_of_the_first_occurrence_in_a_string/strStr.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        n = len(haystack)
        m = len(needle)
        l, r = 0, m - 1

        if m > n:
            return -1

        while l < n:
            if l < n and haystack[l] == needle[0]:
                k = 0
                for i in range(0, m):
                    if i + l < n and haystack[i + l] == needle[i]:
                        k += 1
                if k == m:
                    return l
                l += 1
            else:
                l += 1

        return -1


        
        

File: 28_find_the_index_of_the_first_occurrence_in_a_string/strStr.go
func strStr(haystack string, needle string) int {
    n, m := len(haystack), len(needle)

    for i := 0; i < n - m + 1; i++ {
        if haystack[i: i + m] == needle {
            return i
        }
    }

    return -1
}